Add support for building (x86_64) on Apple Silicon

mbullington opened the pull request:
<!-- Thank you for submitting a pull request! -->
<!-- ⚠️⚠️ Do Not Delete This! pull_request_template ⚠️⚠️ -->
<!-- Please read our contribution guidelines: https://microsoft.github.io/AirSim/CONTRIBUTING/ -->

Fixes: https://github.com/microsoft/AirSim/issues/3696 https://github.com/microsoft/AirSim/issues/3691
<!-- Fixes: # -->
<!-- Fixes: # -->

## About
- This PR adds a solution for the above tickets (which are closed but require out-of-tree script modifications). The goal is for AirSim to compile and run under x84_64 on Apple Silicon machines.
  * Use `brew --prefix` instead of `/usr/local` in case of M1, which resides at `/opt/homebrew`.
  * Specify CMake flag https://cmake.org/cmake/help/latest/variable/CMAKE_APPLE_SILICON_PROCESSOR.html, which basically just tells CMake to use x86_64 (Rosetta) on M1 machines. This allows linking to succeed with the x86 versions of UE and Unity.
- Additionally, in the test project, allow the user to specify `UnrealDir` and have that be used.

## How Has This Been Tested?

This was tested on my local machine, with the following specs:

- M1 MacBook Air
- macOS Monterey
- Unreal Engine 4.27

<!-- Please, describe how you have tested your changes to help us incorporate them. -->

## Screenshots (if appropriate):

mbullington committed:
Add support for building on Apple Silicon

rajat2004 commented on build.sh:
```diff
@@ -94,14 +94,20 @@ if [[ ! -d $build_dir ]]; then
+# Fix for Unreal/Unity using x86_64 (Rosetta) on Apple Silicon hardware.
+CMAKE_VARS=
+if [ "$(uname)" == "Darwin" ]; then
+    CMAKE_VARS="-DCMAKE_APPLE_SILICON_PROCESSOR=x86_64"
```
Doesn't it require CMake 3.19.2 or higher? Maybe `setup.sh` needs to be modified as well, or alteast a note in the MacOS page

mbullington committed:
Bump minimum CMake version on macOS

mbullington commented:
Updated the notes in https://github.com/microsoft/AirSim/pull/4409 and changed the setup to include this bit:

```sh
# On macOS, make sure we have a CMake that will support CMAKE_APPLE_SILICON_PROCESSOR.
if [ "$(uname)" == "Darwin" ]; then
    MIN_CMAKE_VERSION=3.19.2
fi
```

mbullington commented:
Thanks for your quick review!

zimmy87 commented:
All review feedback has been addressed and the checks are passing, so I am moving forward with merging this, thank you for the PR @mbullington!
